dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -81,9 +85,7 @@ private RetryingCallback(Channel delegate, Endpoint endpoint, Request request, i
                     logRetry(failure);
                     return execute();
                 }
-                // TODO(ckozak): It's out of scope for this class to map responses to exceptions. After retries
-                // are exhausted the open response should be returned.
-                return Futures.immediateFailedFuture(new SafeRuntimeException("Retries exhausted", failure));
```

Comment:
:+1: makes sense! users want to see something from the server, not deep in here. Maybe debug log instead?